test/e2e/prometheus_test.go

```diff
@@ -5169,6 +5169,77 @@ func testPrometheusStatusScale(t *testing.T) {
 	}
 }
 
+// TODO: Also make sure no governing service was created?
+// What do we want to do? we want to make sure that when you are creating your own service and
+// the service is actually selected. But this is pretty self explanatory.
+// Something that we also want to check for is to make sure that if the service does not select
+// our prometheus, and the prometheus selects our service then we reject such a config.
```

Comment:
Yes this would useful to test the error cases.

Reply:
Forgot to remove this, mb!